app/services/time_discrepancy_loader.rb

```diff
@@ -1,26 +1,22 @@
 # frozen_string_literal: true
 
 class TimeDiscrepancyLoader
-  DECIMALS_TO_ROUND_TO = 2
-
   # `scope` is an AR collection proxy for TimerSession
   def self.uneven_timer_sessions(scope)
-    scope.joins(:time_entries)
-         .group(:id, :timer_start, :timer_end)
-         .having(
-           <<~SQL
-             CAST(SUM(time_entries.hours)
-             AS DECIMAL(10, #{DECIMALS_TO_ROUND_TO}))
-             !=
-             CAST((#{time_difference_expr}) / 3600.0
-             AS DECIMAL(10, #{DECIMALS_TO_ROUND_TO}))
-           SQL
-         )
+    scope
+      .joins(timer_session_time_entries: :time_entry)
+      .group('timer_sessions.id')
+      .having(
+        <<~SQL
+          timer_sessions.hours != SUM(time_entries.hours)
+          OR timer_sessions.hours != #{time_difference_expr} / 3600
```

Comment:
Why do we need both conditions here?

Reply:
Because we're comparing 3 dates: the time diff between start and end, the summed up hours of each entry and the total hours

Reply:
Ok, I checked this. This seems fine.